src/NUnitEngine/nunit.engine/Runners/DirectTestRunner.cs

```diff
@@ -158,7 +158,8 @@ protected override TestEngineResult RunTests(ITestEventListener listener, TestFi
                 packages.Add(TestPackage);
             foreach (var package in packages)
             {
-                _assemblyResolver.RemovePathFromFile(package.FullName);
+                if (_assemblyResolver != null)
+                    _assemblyResolver.RemovePathFromFile(package.FullName);
```

Comment:
It seems like the if statement can wrap all the code that runs through the packages, starting at line 156.